### PR TITLE
Introducing coarse-grained molecular dynamics simulations and updates to training

### PR DESCRIPTION
**Pull request by datagato**
High-level updates from this PR:
- Introduce coarse grained MD simulations using the Martini 3 force field (with solvent subtraction) for faster Proteogram generation (5-10x faster than atomistic)
- Fixes to atomistic Proteogram creation for more consistent channel logic and stacking
- New training script that does train/test split on-the-fly with seed argument for reproducibility (this can be used for k-fold cross-validation later on by simply changing the seed); also introduces use of focal loss (a flag on the command line) for the training loss function
- PDB download script by date cutoff for future data pulls (when going beyond SCOPe)
- Annotation script updates (want a "GO slim" as a future label for Proteograms for classification according to the Gene Ontology annotations (biological process, molecular function and cellular component)
- Documentation updates

### CODE REVIEW

File: proteogram/v2/martini_nonbonded_forces.py [hunk 1]
```diff
@@ -0,0 +1,1520 @@
...
+    MARTINI_BB_ANGLE_SHEET    as _BB_ANGLE_SHEET,
+    MARTINI_BEAD_MASS_DA      as _BEAD_MASS_DA,
+    MARTINI_WATER_MASS        as _WATER_MASS,
+    MARTINI_ION_MASS_NA       as _ION_MASS_NA,
+    MARTINI_ION_MASS_CL       as _ION_MASS_CL,
+    MARTINI_RESIDUE_BEADS     as _RESIDUE_BEADS,
+)
+
+
+class MartiniNonBondedForceModel:
```
Comment (swapnilashtekar) [inline proteogram/v2/martini_nonbonded_forces.py:118]: nit: There is no shared base class between `AtomisticNonBondedForceModel` and `MartiniNonBondedForceModel`. Each reimplements boilerplate code non-identically and independently. A `BaseNonBondedForceModel` would make a third force-field variant meaningfully cheaper to add later.

File: scripts/v2/create_scope_annotation_file.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
+"""
```
Comment (swapnilashtekar) [inline scripts/v2/create_scope_annotation_file.py:1]: nit: There is duplicated annotation-script logic. `create_scope_annotation_file.py` and `create_pdb_annotation_file.py` share GO/RCSB/UniProt fetch-and-cache logic copy-pasted rather than factored into proteogram/common/utils.py (which both already import from). 
The SCOPe-vs-PDB split itself is well-motivated; the shared plumbing is missing.

File: proteogram/v2/martini_nonbonded_forces.py [hunk 3]
```diff
@@ -0,0 +1,1520 @@
...
+        del diff
+
+        # Apply the same 1.1 nm cutoff as the OpenMM CutoffPeriodic forces.
+        # Without this, the RF term k_rf·r² grows unboundedly beyond the
+        # cutoff, producing spuriously large energies for distant residue pairs
+        # and inflating map intensities relative to atomistic.
+        cutoff_mask = self._bead_valid_mask & (dist_nm < _SOLVENT_CUTOFF_NM)
+        r_safe = np.where(cutoff_mask, dist_nm, 1.0)
+
+        # ── LJ-12-6 (Lorentz-Berthelot combining rules) ───────────────────────
```
Comment (swapnilashtekar) [inline proteogram/v2/martini_nonbonded_forces.py]: As per my understanding CG energy maps use a different force field than the CG dynamics they're sampled from.
`martini_nonbonded_forces.py` builds the actual simulation from an explicit pairwise `σ/ε` lookup table specifically because as per docs, Lorentz-Berthelot combining rules "deviate by up to 9 kJ/mol for key pairs." But `_calc_pairwise_martini_energies` (line 1230), which produces the pixel values written into the proteogram image, recomputes LJ energies from single per-bead-type σ/ε values combined via plain Lorentz-Berthelot, the exact scheme the simulation setup was built to avoid. 

So the trajectory is sampled under one force field, but the output image channels are computed post-hoc under a measurably different, less accurate one. The methodology doc also only documents the Lorentz-Berthelot scheme, so it doesn't match the actual dynamics either.

File: scripts/v2/train_multiple_models_randomized_eval.py [hunk 4]
```diff
@@ -0,0 +1,1562 @@
...
+            pool_indices,
+            test_size=args.val_size,
+            random_state=args.seed,
+            stratify=[labels[i] for i in pool_indices])
+        print(f'Static test list ({os.path.basename(args.test_list)}): '
+              f'{len(test_indices)} test held-out; remaining {len(pool_indices)} split '
+              f'(seed={args.seed}) into {len(train_indices)} train / {len(val_indices)} val '
+              f'(val = {args.val_size:.0%} of remainder; --test_size ignored).')
+    else:
+        train_val_indices, test_indices = train_test_split(
```
Comment (swapnilashtekar) [inline scripts/v2/train_multiple_models_randomized_eval.py:1288]: `train_test_split` is at class level, not at family level. 
Is this intentional? Since `HierarchicalPKSampler` operate at family level.

**Review by swapnilashtekar — CHANGES_REQUESTED** [review]
Overall looks good to me. All comments are non-blocking except [one](https://github.com/datagato/proteogram/pull/9#discussion_r3823509327).

Thank you for creating the change, it is very competent and novel work of CG martini to image channel.

File: proteogram/v2/martini_nonbonded_forces.py (hunk 1 above)
Comment (datagato) [reply proteogram/v2/martini_nonbonded_forces.py:118]: Thanks for the advice!  Let's address this in a PR focused on refactoring.  We can do this next.

File: scripts/v2/create_scope_annotation_file.py (hunk 2 above)
Comment (datagato) [reply scripts/v2/create_scope_annotation_file.py:1]: Again, thanks for the advice!  Let's address this in a PR focused on refactoring.  We can do this next with the one above.

File: scripts/v2/train_multiple_models_randomized_eval.py (hunk 4 above)
Comment (datagato) [reply scripts/v2/train_multiple_models_randomized_eval.py:1288]: I am looking and do see that `train_test_split` stratifies at class by default, but it can also be a command line argument for the training script so can be any level.  Do you mean the default, here?  Thanks!

File: proteogram/v2/martini_nonbonded_forces.py (hunk 3 above)
Comment (datagato) [reply proteogram/v2/martini_nonbonded_forces.py]: This is a really nice catch, thank you so much, and you are spot on!  I am patching this class to look up `σ_ij/ε_ij` from the same per-type-pair tables the simulation LJ force uses (`Discrete2DFunction` in `setup_system`), rather than recombining single-bead σ/ε via Lorentz-Berthelot.  I will test it shortly.
